src/credentials/sharedCredentialsValidation.ts

```diff
@@ -6,89 +6,127 @@
  */
 
 import { localize } from 'vscode-nls'
-import { SectionName, SharedCredentialsKeys, StaticCredentialsProfileData } from './types'
+import {
+    CredentialsData,
+    CredentialsKey,
+    SectionName,
+    SharedCredentialsKeys,
+    StaticCredentialsProfileKeysOptional,
+} from './types'
 import { ToolkitError } from '../shared/errors'
 import { profileExists } from './sharedCredentials'
+import { getLogger } from '../shared/logger'
 
 /**
- * The format validators for shared credentials keys.
+ * A function that validates a credential value
  *
- * A format validator validates the format of the data,
- * but not the validity of the content.
+ * @returns An error message string if there is an error, otherwise undefined.
  */
-export const CredentialsKeyFormatValidators = {
-    [SharedCredentialsKeys.AWS_ACCESS_KEY_ID]: getAccessKeyIdFormatError,
-    [SharedCredentialsKeys.AWS_SECRET_ACCESS_KEY]: getSecretAccessKeyFormatError,
-} as const
+type GetCredentialError = (key: CredentialsKey, value: string | undefined) => string | undefined
 
 /**
- * Holds the error for each key of static credentials data,
- * if it exists. This allows the user to get all the errors
- * at once.
+ * Validates all credentials values from the given input
+ *
+ * @returns Returns the same shape as the input, but the value of each
+ * key is an error message if the original value is invalid,
+ * otherwise undefined
  */
-export type StaticCredentialsErrorResult = {
-    [k in keyof StaticCredentialsProfileData]: string | undefined
+export function getCredentialsErrors(
```

Comment:
This is the head of the change, it validates an object with credentials data. Internally, this calls individual validation functions.